src/admin/components/forms/RowLabel/types.ts

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Data } from '../Form/types';
+
+export type RowLabelFunction = (value: Data, index?: number) => React.ReactNode
```

Comment:
I'd flatten `RowLabelFunction` and `RowLabelComponent` into just a single type, probably `RowLabelComponent`. As far as React is concerned, I don't believe there is a notable difference between these two types. They both end up returning `React.ReactNode`.